p2p/peer.py

```diff
@@ -495,6 +510,46 @@ class LESPeer(BasePeer):
         # TODO: Disconnect if the remote doesn't serve headers.
         self.head_info = cmd.as_head_info(msg)
 
+    async def do_extra_handshake_checks(self) -> None:
+        if self.network_id != MAINNET_NETWORK_ID:
+            return
+        if self.head_info.block_number < DAO_FORK_MAINNET_BLOCK:
+            return
+        try:
+            await self.wait(self._dao_fork_check(), timeout=DAO_FORK_CHECK_TIMEOUT)
+        except TimeoutError:
+            # Logging as INFO because if we start seeing this too often we might need to increase
+            # the timeout used here.
+            self.logger.info("Timed out waiting for DAO fork header from %s", self)
+            raise
+
+    async def _dao_fork_check(self) -> None:
+        # FIXME: For now we only support the pro-fork chain, but we should provide a config option
```

Comment:
This makes me grumbly because I know we have very little intrinsic motivation to fix this later.....  but also the `PeerPool` doesn't have access to the `Chain` so giving it the data it needs to do these checks might be messy.